solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,400 @@
+  type        = string
+  default     = "public"
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid visibility option. Allowed values are 'public', 'private', or 'public-and-private'."
+  }
+}
+
+variable "resource_tags" {
+  description = "List of tags to apply to resources created by this module."
+  type        = list(string)
+  default     = []
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the VSI resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
+}
+
+##############################################################################
+
+
+##############################################################################
+# VPC Variables
+##############################################################################
+
+variable "existing_vpc_id" {
+  description = "The ID of an existing VPC."
+  type        = string
+}
+
+variable "existing_subnet_id" {
+  description = "The ID of an existing subnet"
+  type        = string
+}
+
+##############################################################################
+
+
+##############################################################################
+# VSI Variables
+##############################################################################
+
+variable "vsi_name" {
+  description = "The name of the Virtual server instance."
+  type        = string
+  default     = "vsi"
+}
+
+variable "image_id" {
+  description = "Image ID used for VSI. Run 'ibmcloud is images' to find available images in a region."
+  type        = string
+  default     = "r006-cc341965-a523-464e-969f-391e2661c125"
+}
+
+variable "ssh_public_key" {
+  description = "A public SSH Key for VSI creation which does not already exist in the deployment region. Must be an RSA key with a key size of either 2048 bits or 4096 bits (recommended) - See https://cloud.ibm.com/docs/vpc?topic=vpc-ssh-keys. To use an existing key, enter a value for the variable 'existing_ssh_key_name' instead."
+  type        = string
+  default     = null
+  validation {
+    error_message = "Public SSH Key must be a valid ssh rsa public key."
+    condition     = var.ssh_public_key == null || can(regex("ssh-rsa AAAA[0-9A-Za-z+/]+[=]{0,3} ?([^@]+@[^@]+)?", var.ssh_public_key))
+  }
+  validation {
+    condition     = var.ssh_public_key != null || length(var.existing_ssh_key_ids) > 0 ? true : false
+    error_message = "Please provide a value for either `ssh_public_key` or `existing_ssh_key_ids`."
+  }
+}
+
+variable "existing_ssh_key_ids" {
+  description = "IDs of existing SSH keys to use while creating VSI."
+  type        = list(string)
+  default     = []
+}
+
```

Comment:
Did we say we add a third option to create public / private SSH keys and include them as outputs, and support storing them in secrets manager